python/rmm/rmm/_lib/device_buffer.pyx

```diff
@@ -138,6 +138,30 @@ cdef class DeviceBuffer:
         }
         return intf
 
+    def prefetch(self, device=None, stream=None):
+        """Prefetch buffer data to the specified device on the specified stream.
+
+        Assumes the storage for this DeviceBuffer is CUDA managed memory
+        (unified memory). If it is not, this function is a no-op.
+
+        Parameters
+        ----------
+        device : optional
+            The CUDA device to which to prefetch the memory for this buffer.
+            Defaults to the current CUDA device.
```

Comment:
Q: Maybe this is just a docstring question, but what's the Python story for prefetching on CPU (`cudaCpuDeviceId`)? On the C++ side, we can initialize `rmm::cuda_device_id` to `cudaCpuDeviceId` (which is -1) without problem. 

This is not blocking, just curious about the CPU story since HMM/ATS was brought up in the reviews above.

Reply:
>  but what's the Python story for prefetching on CPU (cudaCpuDeviceId)? On the C++ side

NUMA nodes.

Reply:
Mind a few more words?

Reply:
I don't know how to do the NUMA nodes @jrhemstad mentions, but CUDA Python docs mention that it supports passing `cudaCpuDeviceId` just like in C++. https://nvidia.github.io/cuda-python/module/cudart.html#cuda.cudart.cudaMemPrefetchAsync

Reply:
Ah, now I see there is a `cudaMemPrefetchAsync_v2` of which I was previously unaware that takes more options for `dstLocation`. Notice that this version is undocumented for CUDA Python: https://nvidia.github.io/cuda-python/module/cudart.html#cuda.cudart.cudaMemPrefetchAsync_v2

Reply:
It seems `cudaMemPrefetchAsync_v2` comes without any docstring, so it is undocumented everywhere, let me check internally.

Regardless, I would like to have @rapidsai/rmm-python-codeowners opinions here. I don't think you get my question, so let me rephrase: Do we expect Pythonusers to know that they need to pass -1 here for prefetching CPU? Why? Is it mentioned anywhere that this is the way to go?

Reply:
Whether this is the right UX is subject to discussion, but the minimal work needed is something along this line:
```suggestion
        device : optional
            The CUDA device to which to prefetch the memory for this buffer.
            Defaults to the current CUDA device. To prefetch on CPU, pass ``-1``.
```

Reply:
NACK: "To prefetch on CPU" makes no sense as a sentence, this code is executing on the CPU.

If we want this in the docstring at all, explicitly mention the (exposed by cuda-python) `cudaCpuDeviceId` constant

> To prefetch to the CPU, pass the ``cudaCpuDeviceId`` from cudart as the device.

Reply:
OK. So we're ok to pass a pre-defined constant here. Then I suggest 
1. Add `"cuda-python": ("https://nvidia.github.io/cuda-python/", None),` to this dict:
https://github.com/rapidsai/rmm/blob/f47ce3f0d46848cd9d5844d499bf150dd14d823a/python/rmm/docs/conf.py#L209-L212
2. Update the docstring per Lawrence's suggestion, with cross-referencing to CUDA Python docs:
```suggestion
        device : optional
            The CUDA device to which to prefetch the memory for this buffer.
            Defaults to the current CUDA device. To prefetch to the CPU, pass
            `~cuda.cudart.cudaCpuDeviceId` as the device.
```

Reply:
I was looking at this yesterday. Once 12.2 is our minimum version, we can switch to `_v2` and and modify the prefetching API to accept a locationType parameter. 

Or, I could make an API that does that now, but will only work for NUMA prefetching on 12.2 and later.

Reply:
What is the `~` in your code?

Reply:
`~` would omit the full module qualifier so as to let Sphinx render it to just `cudaCpuDeviceId` (but with the correct hyperlink, for which a full qualifier is needed to compute).

Reply:
Thanks, updated docstring.